### ISSUE TITLE
Coded example box improvements

### ISSUE THREAD

**Issue by murrlipp** [ministryofjustice/moj-frontend#958]
## What
The coded example box template and associated HTML/Nunjucks/Figma tabs that appear on each component page would benefit from changes to make them more usable and clear.

## Why
Whilst working on other tickets, @chrispymm and @murrlipp have identified some improvements to be made.

## Tasks
- [x] Consider making wrapper for 'Open this example in a new window' full width so that examples aren't partially obscured when scrolling in examples.
- [x] Reconsider use of colour when HTML, Nunjucks and Figma tabs are active. There are currently multiple shades of grey that are very similar.
- [x] Reconsider use of grey as a background colour for tabbed content, now that the page background colour has been updated to grey.
- [x] Consider adding ability to set separate example box heights for desktop and mobile, so that example box heights aren't unnecessarily large when viewed on mobile.
- [x] Relay decisions to team.
- [x] Make changes and merge into main.

## Who needs to be involved
- Interaction Designer
- Developer

## Done
When the above tasks have been investigated, decisions made and updated in the appropriate templates.

**Comment by robertjmccarthy** [comment]
Pulled this into Cycle 5. However, priority for the work is low so it should be pulled forward if capacity allows. If not, we'll move this ticket to Cycle 6 for prioritisation.

**Comment by murrlipp** [comment]
@chrispymm here's the branch: https://github.com/ministryofjustice/moj-frontend/tree/update-example-box

I haven't touched the following point:
- [ ] Consider adding ability to set separate example box heights for desktop and mobile, so that example box heights aren't unnecessarily large when viewed on mobile.

And there's a strange styling quirk where a left and right border don't appear when the HTML tab is expanded.  I can't see a reason for it, as discussed on Slack.  If you're able to look at these two points that should be this ticket done.

![Image](https://github.com/user-attachments/assets/5056e9cc-0d05-48f9-9e9c-efc538c3fae2)

**Comment by chrispymm** [comment]
@murrlipp I've pushed a couple of updates to this branch now.

Sorted out the weird gap issue.

I have also added a dynamic iFrameResizer - this watches all changes to the size of the contents of the iframe and adjusts its height automatically.  This accounts for screen size changes, and also changes within the component - i.e. popping up the datepicker.  This means we don't need iframes bigger than necessary - they adjust when things in the component pop up! 🎉 

However this is quite a big change so may be worth previewing to the team to see what they think about it.

May also be worth double-checking with Ben that it doesn't introduce any accessibility concerns too
